src/main/kotlin/org/rust/lang/core/psi/ext/RsFunction.kt

```diff
@@ -22,32 +22,20 @@ import org.rust.lang.core.types.ty.TyUnknown
 import org.rust.lang.core.types.type
 import javax.swing.Icon
 
-private val FUNCTION_TEST_REGEX = Regex("""^[\w:]*test.*""")
-
 val RsFunction.block: RsBlock? get() = PsiTreeUtil.getChildOfType(this, RsBlock::class.java)
 val RsFunction.stubOnlyBlock: RsBlock? get() = PsiTreeUtil.getStubChildOfType(this, RsBlock::class.java)
 
 val RsFunction.isAssocFn: Boolean get() = !hasSelfParameters && owner.isImplOrTrait
 val RsFunction.isMethod: Boolean get() = hasSelfParameters && owner.isImplOrTrait
 
 val RsFunction.isTest: Boolean
-    get() {
-        val stub = greenStub
-        return stub?.isTest
-            ?: (queryAttributes.isTest)
-    }
+    get() = queryAttributes.isTest
 
-val QueryAttributes.isTest
-    get() = hasAttribute(FUNCTION_TEST_REGEX) || hasAtomAttribute("quickcheck")
+private val QueryAttributes.isTest
+    get() = metaItems.mapNotNull { it.path?.referenceName }.any { it.endsWith("test") } || hasAtomAttribute("quickcheck")
```

Comment:
Is this correct? `FUNCTION_TEST_REGEX` allowed items like `footestbar`, but this does not. Do we have any tests on this?

Reply:
@Kobzol looks like `FUNCTION_TEST_REGEX` was introduced by your commit, maybe you have some observations as well?

Reply:
`FUNCTION_TEST_REGEX` was introduced to support cases like `#[tokio::test(flavor = "multi_thread", worker_threads = 1)]`. It looks like iterating through the meta items and finding anything that resembles `test` should be fine.

The corresponding test is `test custom test attribute with underscore`.

Reply:
`it.endsWith("test")` breaks support for custom test frameworks (see https://doc.rust-lang.org/beta/unstable-book/language-features/custom-test-frameworks.html) that currently use `test_case` attribute. I know that we didn't implement anything special for them (I didn't even know about it until recently) and we don't have the corresponding test(s) but anyway

Reply:
Let's use `contains("test")`?